### ISSUE TITLE
Vanilla python image had very old package versions

### ISSUE THREAD

**Issue by wildintellect** [MAAP-Project/ZenHub#955]
**Describe the bug**
I just got a warning 
```
/opt/conda/envs/vanilla/lib/python3.10/site-packages/fsspec/registry.py:273: UserWarning: Your installed version of s3fs is very old and known to cause
severe performance issues, see also https://github.com/dask/dask/issues/10276

To fix, you should specify a lower version bound on s3fs, or
update the current installation.
`
Checking the package versions 0.4.2
```
conda list | grep s3fs
s3fs                      0.4.2                      py_0    conda-forge
```
This version is more than 3 years old.
https://anaconda.org/conda-forge/s3fs/files

**To Reproduce**
Steps to reproduce the behavior:
1. Use pandas to open a csv from s3

This is `Vanilla 3.1.4`

**Comment by wildintellect** [comment]
It's pinned to that version, 
https://github.com/MAAP-Project/maap-workspaces/blob/f1a395a1ea2b35168806961a3d385ce3edea0fe9/base_images/vanilla/environment.yml#L37

**Comment by wildintellect** [comment]
The Pangeo image has `s3fs=2023.12.2` so at least this version should be safe.
We need to review the process of updating package versions to make sure all the pins are included.

**Comment by grallewellyn** [comment]
I changed all versions to be 2023.12.2
I agree, it makes sense to update all the packages at once and I think if we are in the scenario that a package needs to be updated, the developer would just increase the version for all instances of that package (ie you specifically called out vanilla but I also updated isce3 to the same version in my PR) 

I think the bigger question would be how do we go about knowing to update package versions. If someone points out the version is very old like you did, that is an easy find. But how would we do it in a normal case? Maybe every year or so run a `conda update --all` and work out the any errors? We are hoping to write some tests for all our installed packages and use papermill to automate this testing process 
cc @anilnatha
